app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -52,30 +56,60 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
     db.collection(collectionPath)
         .get()
         .addOnSuccessListener { querySnapshot ->
-          Log.d("UserRepositoryFirestore", "getAllUsers")
           val users =
               querySnapshot.documents.mapNotNull { document ->
-                Log.d("UserRepositoryFirestore", document.data.toString())
-                document.data?.let { deserializeUser(it) }
+                val userPublic = deserializeUser(document.data!!)
+                User(userPublic, null)
               }
           onSuccess(users)
         }
         .addOnFailureListener { onFailure(it) }
   }
 
   override fun addUser(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(user.userId)
-        .set(serializeUser(user))
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { onFailure(it) }
+    if (user.details == null) {
+      onFailure(Exception("User details are required"))
+      return
+    }
+    db.collection(collectionPath).document(user.public.userId).get().addOnSuccessListener {
```

Comment:
Please don't forget to add the onFailure listener to this database call as well!